File: latest-code-output/clarity_ritual_bootstrap_pack/veilLog.js
// veilLog.js — Standard ritual log exporter for Node.js

const fs = require('fs');
const path = './ritual_manifest/spells.log.json';

function veilLog(entry) {
  const timestamp = new Date().toISOString();
  const record = { timestamp, ...entry };

  let log = [];
  if (fs.existsSync(path)) {
    log = JSON.parse(fs.readFileSync(path, 'utf8'));
  }
  log.push(record);
  fs.writeFileSync(path, JSON.stringify(log, null, 2));
  // Optionally print for local echo:
  console.log(`[VEIL LOG] ${timestamp}:`, entry);
}

// Usage:
// veilLog({ spell: "Entropy Scanner", result: "complete", notes: "Scan cycle closed" });

module.exports = { veilLog };

File: latest-code-output/clarity_ritual_bootstrap_pack/veil_log.py
# veil_log.py — Standard ritual log exporter for Python

import json, os
from datetime import datetime

LOG_PATH = './ritual_manifest/spells.log.json'

def veil_log(spell, result, notes=None):
    record = {
        "timestamp": datetime.utcnow().isoformat() + "Z",
        "spell": spell,
        "result": result,
        "notes": notes or ""
    }
    log = []
    if os.path.exists(LOG_PATH):
        with open(LOG_PATH) as f:
            log = json.load(f)
    log.append(record)
    with open(LOG_PATH, 'w') as f:
        json.dump(log, f, indent=2)
    print(f"[VEIL LOG] {{record['timestamp']}}: {{spell}} – {{result}}")

# Usage:
# veil_log("Entropy Scanner", "complete", "Scan cycle closed")
